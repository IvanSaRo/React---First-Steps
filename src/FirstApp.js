// Functional component

import React from "react";
import PropTypes from 'prop-types';




const FirstApp = ( {saludo, subtitle}  /* = 'Soy un subtitulo' */ ) => {

    
  return (
    <>
      
      <h2> {saludo} </h2>
     
      <p> {subtitle}</p>
    </>
  );
};

 FirstApp.propTypes = {
    saludo: PropTypes.string.isRequired
} 

FirstApp.defaultProps = {
    subtitle: 'Soy un subtitulo'
}
export default FirstApp;

