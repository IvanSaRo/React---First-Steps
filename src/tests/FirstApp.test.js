import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import FirstApp from "../FirstApp";

describe("Pruebas en FirstApp", () => {
  test("Debe mostrar FirstApp correctamente", () => {
    const saludo = "Hola, soy Iván";
    const wrapper = shallow(<FirstApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot(); /*  */
  });

  test('Debe mostrar el subtitulo enviado por props', () => {
    const saludo = "Hola, soy Iván";
    const subTitle = "Soy un subtitulo";
    
    const wrapper = shallow(
        <FirstApp 
            saludo={saludo} 
            subtitle= {subTitle}
        />
    );
    
    const textoParrafo = wrapper.find('p').text();

    expect(textoParrafo).toBe(subTitle);
  })
  
});
