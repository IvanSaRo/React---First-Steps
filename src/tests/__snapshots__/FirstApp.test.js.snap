// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe mostrar FirstApp correctamente 1`] = `
<Fragment>
  <h2>
     
    Hola, soy Iv√°n
     
  </h2>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
